src/2_branch.js

```diff
@@ -139,182 +172,184 @@ Branch.prototype['createLink'] = function(obj, callback) {
-	self.api.makeRequest(config.resources.links.createLinkClick, {
-		url: url
-	}, function(data) {
-		if (typeof callback == 'function') { callback(data.click_id); }
+/**
+ * @param {?String} url
+ * @param {?function} callback
+ */
+Branch.prototype['createLinkClick'] = function(url, callback) {
+	callback = callback || function() {};
+	if (!this.initialized) { return utils.console(utils.messages.nonInit); }
+	
+	this.api(resources.createLinkClick, {
+		link_url: url.replace('https://bnc.lt/', ''),
+		click: "click"
+	}, function(err, data) {
+		if (err) { callback(err); }
+		else { callback(null, data); }
 	});
 };
-this.SMSLink = function(obj, callback) {
-	if (!self.initialized) { return utils.console(config.debugMsgs.nonInit); }
-	var phone = obj.phone;
-	obj.channel = 'sms';
-	if (config.linkId === undefined) {
-		this.createLink(obj, function(url) {
-			self.api.makeRequest(config.resources.links.createLinkClick, {
-				link_url: url.replace(/^http:\/\/[^\/]+/, 'https://bnc.lt') + '?click'
-			}, function(data) {
-				self.sendSMSLink(phone, config.linkUrl + '/c/' + data.click_id, function() {
-					if (typeof callback == 'function') { callback(); }
+
+/**
+ * @param {?Object} metadata
+ * @param {?function} callback
+ */
+Branch.prototype['SMSLink'] = function(obj, callback) {
+	callback = callback || function() {};
+	if (!this.initialized) { return utils.console(utils.messages.nonInit); }
+	obj["channel"] = 'sms';
+	var self = this;
+	this.createLink(obj, function(url) {
+		self.createLinkClick(url, function(err, data) {
+			if (err) { callback(err); }
+			else {
+				self.sendSMSLink(obj["phone"], data, function(data) {
+					if (err) { callback(err); }
+					else { callback({}); }
 				});
-			});
-		});
-	}
-	else {
-		self.sendSMSLink(phone, config.linkUrl + '/c/' + config.linkId, function() {
-			if (typeof callback == 'function') { callback(); }
+			}
 		});
-	}
+	});
 };
-this.sendSMSLink = function(phone, url, callback) {
-	self.api.makeRequest(config.resources.links.sendSMSLink, {
-		link_url: url.replace(/^http:\/\/[^\/]+/, 'https://bnc.lt'), // Always go to HTTPS.
+
+/**
+ * @param {?String} phone
+ * @param {?Object} data
+ * @param {?function} callback
+ */
+Branch.prototype['sendSMSLink'] = function(phone, data, callback) {
+	callback = callback || function() {};
+	this.api(resources.sendSMSLink, {
+		link_url: data.click_id,
 		phone: phone
-	}, function(data) {
-		if (typeof callback == 'function') { callback(data); }
+	}, function(err, data) {
+		callback(err, data);
 	});
 };
-this.showReferrals = function(callback) {
-	if (!self.initialized) { return utils.console(config.debugMsgs.nonInit); }
-	self.api.makeRequest(config.resources.referrals.showReferrals, {
-		app_id: config.appId,
-		identity_id: utils.identity()
-	}, function(data) {
-		if (typeof callback == 'function') { callback(data); }
+
+/**
+ * @param {?function} callback
+ */
+Branch.prototype["showReferrals"] = function(callback) {
```

Comment:
we can probalby rename a bunch of these - like this would probably just be better called referrals, the next would probably be better called credits, etc. The goal is to make this feel as idiomatic as possible for people using the library – it needs to feel like any other JS library